etc/deltaDebug.py

```diff
@@ -0,0 +1,304 @@
+################################
+# You need to provide the script with 3 main arguments
+# --base_db_path <the relative path to the db file to perform the step on>
+# --error_string <the output that indicates a target error has occured>
+# --step <Command used to perform a step on the base_db file>
+# You also have 1 additional argument
+# --persistence <a value in [1,6] indicating maximum granularity where maximum granularity = 2^persistence>
+
+# EXAMPLE COMMAND:
+# Assuming running in a directory with the following files in:
+# deltaDebug.py base.odb step.sh
+# openroad -python deltaDebug.py --base_db_path base.odb --error_string <any_possible_error> --step './step.sh' 
+#                                --use_stdout 1 --persistence 5 --dump_def 0
+
+# N.B: step.sh shall read base.odb (or base.def in case the flag dump_def = 1) and operate on it
+# where the script manipulates base.odb between steps to reduce its size. 
+################################
+
+import odb
+import os
+import signal
+import subprocess 
+import argparse
+import shutil
+import select
+import time
+from math import ceil
+import errno
+import enum
+
+
+
+parser = argparse.ArgumentParser('Arguments for delta debugging')
+parser.add_argument('--base_db_path', type=str, help='Path to the db file to perform the step on')
+parser.add_argument('--error_string', type=str, help='The output that indicates target error has occured')
+parser.add_argument('--step', type=str, help='Command used to perform step on the input odb file')
+parser.add_argument('--persistence', type=int, default=1, choices=[1,2,3,4,5,6], help= 'Indicates maximum input fragmentation; fragments = 2^persistence; value in [1,6]')
+
+dump_def = 0    # a flag to indicate whether to dump def per each step, could be used if the step acts on a def file rather than an odb
+use_stdout = 0  # a flag either 0 or 1 to enable/disable detecting the error string from stdout in addition to stderr
```

Comment:
Why aren't these command line args anymore?  I mean you should use 

parser.add_argument('--use_stdout', action='store_true')

Reply:
Sorry, miss understood what you meant by this edit. Fixed.